jungfrau_gui/ui_components/tem_controls/task/task_manager.py

```diff
@@ -93,84 +95,110 @@ def __init__(self, tem_action): #, timeout:int=10, buffer=1024):
+        self.task_thread = QThread()  
 
         self.tem_action.parent.threadWorkerPairs.append((self.task_thread, self.task))
 
-        # why the two layers? TODO: IMPROVE
         self.task.finished.connect(self.on_task_finished)
-        # self.finished_record_task.connect(self.stop_task)
 
         self.task.moveToThread(self.task_thread)
         self.task_thread.started.connect(self.task.start.emit)
         self.task_thread.start()
 
-        if isinstance(self.task, BeamFitTask):
-            self.sweepingWorkerReady = True
-
     @Slot(str)
     def getteminfo(self, gui=''):
         logging.info("Start GetInfo")
         if self.task is not None:
             if self.task.running:
+                logging.warning("Stopping the currently running  - \033[38;5;214mGetInfoTask\033[33m - task before starting a new one.")
                 self.stop_task()
-        # self.send.emit("stage.Setf1OverRateTxNum(2)")
+                return
+
         command='TEMstatus'
+
         if gui=='':
             x = input(f'Write TEM status on a file? If YES, give a filename or "Y" ({command}_[timecode].log). [N]\n')
             task = GetInfoTask(self, x)
         else:
             task = GetInfoTask(self, gui)
+
         self.start_task(task)
 
     @Slot()
     def start_record(self):
-        logging.info("Start Rotation/Record")
+        logging.info("Starting Rotation/Record")
+
+        # Check if a task is already running, and stop it if so
         if self.task is not None:
             if self.task.running:
-                self.stop_task()
+                logging.warning("\033[38;5;214mRecordTask\033[33m - task is currently running...\n"
+                                "You need to stop the current task before starting a new one.")
+                # self.stop_task()  # Ensure that the current task is fully stopped
+                return
+
         end_angle = self.tem_action.tem_tasks.update_end_angle.value() # 60
         logging.info(f"End angle = {end_angle}")
 
         self.file_operations.update_base_data_directory() # Update the GUI
         filename_suffix = self.cfg.data_dir / 'RotEDlog_test'
 
         if self.tem_action.tem_tasks.withwriter_checkbox.isChecked():
-            task = RecordTask(self, end_angle, filename_suffix, writer_event = self.tem_action.file_operations.toggle_hdf5Writer)
+            task = RecordTask(self, end_angle, filename_suffix.as_posix(), writer_event = self.tem_action.file_operations.toggle_hdf5Writer)
         else:
-            task = RecordTask(self, end_angle, filename_suffix)
+            task = RecordTask(self, end_angle, filename_suffix.as_posix())
         self.start_task(task)
 
     @Slot()
     def start_beam_fit(self):
         logging.info("Start AutoFocus")
+
         if self.task is not None:
             if self.task.running:
-                logging.warning('task already running')
+                logging.warning("\033[38;5;214mAutoFocus\033[33m - task is currently running...\n"
```

Comment:
Have a look at [rich](https://github.com/Textualize/rich) it's quite a nice library for printing